main/src/ca/uwaterloo/flix/language/phase/Stratifier.scala

```diff
@@ -804,10 +811,10 @@ object Stratifier extends Phase[Root, Root] {
   /**
     * Returns a path that forms a cycle with the edge from `src` to `dst` in the given dependency graph `g`.
     */
-  private def findNegativeCycle(src: Name.Pred, dst: Name.Pred, g: DependencyGraph, loc: SourceLocation): List[(Name.Pred, SourceLocation)] = {
+  private def findNegativeCycle(src: Name.Pred, dst: Name.Pred, g: Set[DependencyEdge], loc: SourceLocation): List[(Name.Pred, SourceLocation)] = {
```

Comment:
I would keep the old definition of DependencyGraph. The idea would be to leave the old infrastructure, which computes the stratification, in tact. Then it can also be better reused for other implementations.